src/Features/CSharp/Portable/Completion/CompletionProviders/ImportCompletion/ExtensionMethodImportCompletionProvider.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    internal sealed class ExtensionMethodImportCompletionProvider : AbstractExtensionMethodImportCompletionProvider
+    {
+        internal override bool IsInsertionTrigger(SourceText text, int characterPosition, OptionSet options)
+            => ImportCompletionProviderHelper.IsInsertionTrigger(text, characterPosition, options);
+
+        protected override ImmutableArray<string> GetImportedNamespaces(
+            SyntaxNode location,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+            => ImportCompletionProviderHelper.GetImportedNamespaces(location, semanticModel, cancellationToken);
+
+        protected override Task<SyntaxContext> CreateContextAsync(Document document, int position, CancellationToken cancellationToken)
+            => ImportCompletionProviderHelper.CreateContextAsync(document, position, cancellationToken);
+
+        protected override Task<bool> IsInImportsDirectiveAsync(Document document, int position, CancellationToken cancellationToken)
+            => ImportCompletionProviderHelper.IsInImportsDirectiveAsync(document, position, cancellationToken);
+
+        protected override bool TryGetReceiverTypeSymbol(SyntaxContext syntaxContext, out ITypeSymbol receiverTypeSymbol)
+        {
+            if (syntaxContext.TargetToken.Parent is MemberAccessExpressionSyntax memberAccess)
```

Comment:
so if i had `(1 + 1).` then i wouldn't get a receiver type?

Reply:
fixed

Reply:
id don't see how this is fixed.  

1. you still call GetSymbolInfo, even though `(1 + 1)` shouldn't result in a symbol.
2. you still filter down to named types and type-params.  We shouldn't do that.